### PR TITLE
Don't advertise unimplemented features

### PR DESCRIPTION
**Pull request by jaydoane**
## Overview

Removes the following features from the welcome message:
- reshard
- partitioned
- pluggable-storage-engines
- access-ready
- scheduler

## Testing recommendations

```
make check-fdb
```

## Checklist

- [x] Code is written and works correctly
- [x] Changes are covered by tests
- [ ] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [ ] A PR for documentation changes has been made in https://github.com/apache/couchdb-documentation

### CODE REVIEW

File: src/couch/src/couch_server.erl [hunk 1]
```diff
@@ -238,13 +238,16 @@ init([]) ->
...
+    % Not implemented on FDB
+    % config:enable_feature('pluggable-storage-engines'),
 
     % Mark partitioned databases as a supported feature
-    config:enable_feature(partitioned),
+    % Not implemented on FDB
+    % config:enable_feature(partitioned),
 
     % Mark being able to receive documents with an _access property as a supported feature
-    config:enable_feature('access-ready'),
```
Comment (nickva) [inline src/couch/src/couch_server.erl]: We are _access ready technically so let's keep this one enabled. Though couch_server is probably the wrong place for it. We'd want it in fabric2_server eventually or other such place.

https://github.com/apache/couchdb/blob/prototype/fdb-layer/src/couch/src/couch_doc.erl#L304

**Comment by nickva** [comment]
Also perhaps instead of commenting those out we just remove the feature set lines. Except for scheduler since we intend to bring it back.

Another crazy idea is to not start couch_server at all. I am guessing this will breaking a whole bunch of things that try to open system databases and such but if you're curious and want to see what happens that will auto-magically disable all these features. If that works we would want to move access-ready to another place then.

File: rel/overlay/etc/vm.args [hunk 2]
```diff
@@ -60,3 +60,6 @@
 
 # Set maximum SSL session lifetime to reap terminated replication readers
 -ssl session_lifetime 300
+
+# reshard unsupported on FDB
+-mem3 reshard_disabled true
```
[the remark this answers is not in the document]
Comment (jaydoane) [reply rel/overlay/etc/vm.args]: Done.

File: src/couch/src/couch_server.erl (hunk 1 above)
Comment (jaydoane) [reply src/couch/src/couch_server.erl]: Updated.

**Comment by jaydoane** [comment]
> Also perhaps instead of commenting those out we just remove the feature set lines. Except for scheduler since we intend to bring it back.
> 
> Another crazy idea is to not start couch_server at all. I am guessing this will breaking a whole bunch of things that try to open system databases and such but if you're curious and want to see what happens that will auto-magically disable all these features. If that works we would want to move access-ready to another place then.

I commented out the couch_server child spec in couch_primary_sup.erl, but couch layer failed to start:

```erlang
{"init terminating in do_boot",{{error,{{shutdown,{failed_to_start_child,mem3_nodes,{badarg,[{ets,lookup,[couch_dbs_locks,<<"_nodes">>],[]},{couch_server,validate_open_or_create,2,[{file,"src/couch_server.erl"},{line,711}]},{couch_server,open,2,[{file,"src/couch_server.erl"},{line,83}]},{couch_db,open,2,[{file,"src/couch_db.erl"},{line,160}]},{mem3_util,ensure_exists,1,[{file,"src/mem3_util.erl"},{line,270}]},{mem3_nodes,initialize_nodelist,0,[{file,"src/mem3_nodes.erl"},{line,94}]},{mem3_nodes,init,1,[{file,"src/mem3_nodes.erl"},{line,45}]},{gen_server,init_it,2,[{file,"gen_server.erl"},{line,374}]}]}}},{mem3_app,start,[normal,[]]}}},[{boot_node,start_app,3,[{file,"dev/boot_node.erl"},{line,146}]},{lists,foldl,3,[{file,"lists.erl"},{line,1263}]},{boot_node,start_app,3,[{file,"dev/boot_node.erl"},{line,136}]},{lists,foldl,3,[{file,"lists.erl"},{line,1263}]},{boot_node,start_app,3,[{file,"dev/boot_node.erl"},{line,136}]},{lists,foldl,3,[{file,"lists.erl"},{line,1263}]},{boot_node,start_app,3,[{file,"dev/boot_node.erl"},{line,136}]},{lists,foldl,3,[{file,"lists.erl"},{line,1263}]}]}}
```

I think it's going to probably require a bit more work to turn it off, which I presume is out of scope for this commit.

**Pull request merged**